src/appleseed/renderer/modeling/bsdf/bsdf.cpp

```diff
@@ -97,25 +97,45 @@ void* BSDF::evaluate_inputs(
 }
 
 void BSDF::prepare_inputs(
-    Arena&                  arena,
-    const ShadingPoint&     shading_point,
-    void*                   data) const
+    Arena&                      arena,
+    const ShadingPoint&         shading_point,
+    void*                       data) const
 {
 }
 
 float BSDF::sample_ior(
-    SamplingContext&        sampling_context,
-    const void*             data) const
+    SamplingContext&            sampling_context,
+    const void*                 data) const
 {
     return 1.0f;
 }
 
 void BSDF::compute_absorption(
-    const void*             data,
-    const float             distance,
-    Spectrum&               absorption) const
+    const void*                 data,
+    const float                 distance,
+    Spectrum&                   absorption) const
 {
     absorption.set(1.0f);
 }
 
+void BSDF::attenuate_substrate(
+    const void*                 data,
+    const Basis3f&              shading_basis,
+    const Vector3f&             outgoing,
+    const Vector3f&             incoming,
+    Spectrum&                   value) const
+{
+    assert(false);
```

Comment:
Maybe this should be a bit more descriptive. If I understand correctly, we should never reach this assertion unless a BRDF that wasn't designed for layering is used, right? If so, what about something along these lines:

    assert(!"This BSDF does not support layering.");